internal/services/apimanagement/api_management_policy_fragment_resource.go

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apimanagement
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2022-08-01/apimanagementservice"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2022-08-01/policyfragment"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+func resourceApiManagementPolicyFragment() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceApiManagementPolicyFragmentCreate,
+		Read:   resourceApiManagementPolicyFragmentRead,
+		Update: resourceApiManagementPolicyFragmentUpdate,
+		Delete: resourceApiManagementPolicyFragmentDelete,
+		Importer: pluginsdk.ImporterValidatingResourceIdThen(func(id string) error {
+			_, err := policyfragment.ParsePolicyFragmentID(id)
+			return err
+		}, func(ctx context.Context, d *pluginsdk.ResourceData, meta interface{}) ([]*pluginsdk.ResourceData, error) {
+			client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+
+			id, err := policyfragment.ParsePolicyFragmentID(d.Id())
+			if err != nil {
+				return nil, err
+			}
+
+			resp, err := client.Get(ctx, *id, policyfragment.GetOperationOptions{
+				Format: pointer.To(policyfragment.PolicyFragmentContentFormatXml),
+			})
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return nil, fmt.Errorf("Api Management Policy Fragment %q was not found in Api Management instance %q in Resource Group %q", id.PolicyFragmentName, id.ServiceName, id.ResourceGroupName)
```

Comment:
```suggestion
					return nil, fmt.Errorf("%s was not found in Api Management instance %q in Resource Group %q", id, id.PolicyFragmentName, id.ServiceName, id.ResourceGroupName)
```

Reply:
changed it to
```go
return nil, fmt.Errorf("%q was not found in Api Management instance %q in Resource Group %q", id.PolicyFragmentName, id.ServiceName, id.ResourceGroupName)
```
because the id.PolicyFragmentName was already provided as the first parameter and changed the first placeholder from %s to %q.
Hope that is ok or did you explicitly want to use %s and hand in the whole id object?

Reply:
Majority of the `id` objects have an inherent String function that outputs the ID information in a human readable format, so in this case printing the `id` object using using `%s` would output something like:
```
Subscription: "000-000-00000-000000"
Resource Group Name: "resource-group-example"
Service Name: "apim-service-example"
Policy Fragment: "policy-fragment-example"
```

It's not a huge deal, but it provides a little more information in a way that's common across the provider.

Reply:
I see, thanks for clarifying, that sounds good. So then I'd do the following, as you suggested but remove the `id.PolicyFragmentName` since its one arg too many compared to the placeholders:

```go
return nil, fmt.Errorf("%s was not found in Api Management instance %q in Resource Group %q", id, id.ServiceName, id.ResourceGroupName)
```